Zend/tests/partial_function/object-constructor.phpt

```diff
@@ -0,0 +1,85 @@
+--TEST--
+Object/partial interaction, constructors.
+--FILE--
+<?php
+// Constructors
+class Foo {
+    public function __construct($one, $two) {
+        printf(
+            "Rocking %s [%d, %d]\n",
+            __METHOD__, $one, $two);
+    }
+}
+
+$foo = new Foo(?, ?);
+
+var_dump($foo);
+var_dump($first = $foo(1,2));
+var_dump($second = $foo(3,4));
+var_dump($first !== $second);
+
+// Empty constructors
+class Bar {
+
+}
+
+$bar = new Bar(?);
+var_dump($bar, $bar(1,2));
+
+// Failing constructors
+
+class Tantrum {
+    public function __construct(int $one, int $two) {
+        throw new \RuntimeException("I don't wanna.");
+    }
+
+    public function __destruct() {
+        print "This shouldn't happen" . PHP_EOL;
+    }
+}
+
+$p = new Tantrum(?, ?);
```

Comment:
That Closure also falls into trace.

```
class Bar {
    public function __construct($x, $y) {
        throw new Exception('test');
    }
}

try {
    $bClosure = new Bar(?, 2);
    $b = $bClosure(1);
} catch (Exception $e) {
    echo $e->getTraceAsString();
}
```
outputs:
```
#0 .../partials.php(12): Bar->__construct(1, 2)
#1 .../partials.php(12): Closure->__invoke(1)
#2 {main}
```
but _partials_ should be transparent, no?

Reply:
No, they still produce an interstitial Closure object.  But they produce only one interstitial, no matter how many times you partially apply the function.